src/DateTimeParser.jl

```diff
@@ -0,0 +1,550 @@
+module DateTimeParser
+
+using Base.Dates
+using TimeZones
+import Base.Dates: VALUETODAYOFWEEK, VALUETODAYOFWEEKABBR, VALUETOMONTH, VALUETOMONTHABBR
+export parse, tryparse
+
+
+function Base.tryparse(::Type{ZonedDateTime}, str::AbstractString; args...)
+    try
+        return Nullable{ZonedDateTime}(parse(ZonedDateTime, str; args...))
+    catch
+        return Nullable{ZonedDateTime}()
+    end
+end
+function Base.tryparse(::Type{DateTime}, str::AbstractString; args...)
+    try
+        return Nullable{DateTime}(parse(DateTime, str; args...))
+    catch
+        return Nullable{DateTime}()
+    end
+end
+function Base.tryparse(::Type{Date}, str::AbstractString; args...)
+    try
+        return Nullable{Date}(parse(Date, str; args...))
+    catch
+        return Nullable{Date}()
+    end
+end
+
+Base.parse(::Type{ZonedDateTime}, str::AbstractString; args...) = parsedate(str; args...)
+Base.parse(::Type{DateTime}, str::AbstractString; args...) = DateTime(parsedate(str; args...))
+Base.parse(::Type{Date}, str::AbstractString; args...) = Date(DateTime(parsedate(str; args...)))
```

Comment:
Tests are currently breaking due to `DateTime(zdt::ZonedDateTime)` not being defined. Use `import TimeZones: localtime` for now as this is available in the latest release and does what you want.